server/metadata.go

```diff
@@ -571,6 +575,36 @@ func (m *metadataAPI) Reset() error {
 	return nil
 }
 
+// CloseStream close a streams and clears corresponding state in the metadata
+// store.
+func (m *metadataAPI) CloseAndDeleteStream(stream *stream) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	err := stream.Close()
+	if err != nil {
+		return errors.Wrap(err, "failed to delete stream")
+	}
+
+	streamDatapath := filepath.Join(m.Server.config.DataDir, "streams", stream.name)
+	err = os.RemoveAll(streamDatapath)
+	if err != nil {
+		return errors.Wrap(err, "failed to delete stream data path")
+	}
```

Comment:
Here's what I'm thinking we should do instead for deleting. `commitLog` already has a [`Delete` function](https://github.com/liftbridge-io/liftbridge/blob/master/server/commitlog/commitlog.go#L422) which closes and deletes the log, it's just not exposed on the [`CommitLog` interface](https://github.com/liftbridge-io/liftbridge/blob/master/server/commitlog/interface.go). Rather than reimplementing that logic, I think we should expose `Delete` on the interface, add a `Delete` function to `stream`, and call that here.

Reply:
Actually, I misspoke. `Delete` is already [exposed on the interface](https://github.com/liftbridge-io/liftbridge/blob/master/server/commitlog/interface.go#L7).

Reply:
Should be fixed in 89b52f1.